File: LEETCODE/Move_Zeroes/solution_twopointer.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        slow = 0  # 가장 앞쪽의 0의 인덱스를 가리킨다.
        for fast in range(len(nums)):

            # fast가 0이 아닐때만 가장 앞쪽의 0과 스왑한다.
            if nums[fast] != 0:
                nums[fast], nums[slow] = nums[slow], nums[fast]

            # 다음 slow를 찾아간다.
            if nums[slow] != 0:
                slow += 1

File: LEETCODE/Move_Zeroes/solution_twopointer.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
  let slow = 0;
  for (let fast = 0; fast < nums.length; fast++) {
    if (nums[fast] !== 0) {
      const temp = nums[fast];
      nums[fast] = nums[slow];
      nums[slow] = temp;
    }

    if (nums[slow] !== 0) slow++;
  }
}
